agent/ls.go

```diff
@@ -85,7 +89,18 @@ func listFiles(query LSRequest) (LSResponse, error) {
 	}
 
 	if !stat.IsDir() {
-		return LSResponse{}, xerrors.Errorf("path %q is not a directory", absolutePathString)
+		// `ls` on a file should return just that file.
```

Comment:
If we really want to do this the API should be different, like perhaps an `IsDir` on the response type. Since we don't need this for Coder Desktop I think this should just return an error for now.

Reply:
In the future if we need a single file API we would probably add a different endpoint like `/api/v0/stat` that returned more information anyways.